### PR TITLE
Add type hints to decorators module

### PR DESCRIPTION
**Pull request by njgheorghita**
### What was wrong?
Decorators module needed type-hints

#### Cute Animal Picture
image

### CODE REVIEW

File: eth_utils/decorators.py [hunk 1]
```diff
@@ -1,69 +1,76 @@
...
 )
 
 
 class combomethod(object):
-    def __init__(self, method):
+    def __init__(self, method: Any) -> None:
         self.method = method
 
-    def __get__(self, obj=None, objtype=None):
+    def __get__(self, obj: Any=None, objtype: Any=None) -> Any:
```
Comment (pipermerriam) [inline eth_utils/decorators.py]: I *think* `objtype` should be `typing.Type`.

File: eth_utils/decorators.py [hunk 2]
```diff
@@ -1,69 +1,79 @@
...
+)
+
 from .types import (
     is_text,
 )
 
 
 class combomethod(object):
-    def __init__(self, method):
+    def __init__(self, method: Any) -> None:
```
Comment (pipermerriam) [inline eth_utils/decorators.py]: method should be a `Callable[..., Any]`

File: eth_utils/decorators.py [hunk 3]
```diff
@@ -13,63 +15,65 @@
...
-    vals = itertools.chain(args, kwargs.values())
+def _has_one_val(*args: Iterable[Any], **kwargs: Dict[str, Any]) -> bool:
+    # Ignored because mypy cannot infer arg 1 of "chain"
+    vals = itertools.chain(args, kwargs.values())  # type: ignore
     not_nones = list(filter(lambda val: val is not None, vals))
     return len(not_nones) == 1
 
 
-def _assert_one_val(*args, **kwargs):
+def _assert_one_val(*args: Iterable[Any], **kwargs: Dict[str, Any]) -> None:
```
Comment (cburgdorf) [inline eth_utils/decorators.py]: I think a lot of these could use generics to make sure the inner type doesn't get lost to `Any`

**Comment by njgheorghita** [comment]
closing this b/c #177

**Pull request closed without merging**